Core/include/Acts/TrackFinding/ComposableTrackStateCreator.hpp

```diff
@@ -0,0 +1,415 @@
+  /// calibrate measurements using track information, this could be e.g. sagging
+  /// for wires, module deformations, etc.
+  Calibrator calibrator{DelegateFuncTag<
+      detail::voidFitterCalibrator<TrackStateContainerBackend>>{}};
+
+  MeasurementSelector measurementSelector{
+      DelegateFuncTag<voidMeasurementSelector>{}};
+
+  /// Create track states for selected measurements given by the source links
+  ///
+  /// @param gctx The current geometry context
+  /// @param calibrationContext pointer to the current calibration context
+  /// @param surface the surface the sourceLinks are associated to
+  /// @param boundState Bound state from the propagation on this surface
+  /// @param slBegin Begin iterator for sourceLinks
+  /// @param slEnd End iterator for sourceLinks
+  /// @param prevTip Index pointing at previous trajectory state (i.e. tip)
+  /// @param bufferTrajectory a buffer for temporary candidate track states
+  /// @param trackStateCandidates a buffer for temporary track state proxies for candidates
+  /// @param trajectory the trajectory to which new track states for selected measurements will be added
+  /// @param logger the logger for messages.
+  Result<CkfTypes::BranchVector<TrackIndexType>> createSourceLinkTrackStates(
+      const GeometryContext& gctx, const CalibrationContext& calibrationContext,
+      [[maybe_unused]] const Surface& surface, const BoundState& boundState,
+      source_link_iterator_t slBegin, source_link_iterator_t slEnd,
+      TrackIndexType prevTip, TrackStateContainerBackend& bufferTrajectory,
+      std::vector<TrackStateProxy>& trackStateCandidates,
+      TrackStateContainerBackend& trajectory, const Logger& logger) const {
+    using PM = TrackStatePropMask;
+
+    using ResultTrackStateList =
+        Acts::Result<CkfTypes::BranchVector<TrackIndexType>>;
+    ResultTrackStateList resultTrackStateList{
+        CkfTypes::BranchVector<TrackIndexType>()};
+    const auto& [boundParams, jacobian, pathLength] = boundState;
+
+    trackStateCandidates.clear();
+    if constexpr (std::ranges::random_access_range<source_link_iterator_t>) {
+      trackStateCandidates.reserve(std::distance(slBegin, slEnd));
+    }
+
+    // Calibrate all the source links on the surface since the selection has
+    // to be done based on calibrated measurement
+    for (auto it = slBegin; it != slEnd; ++it) {
+      // get the source link
+      const auto sourceLink = *it;
+
+      // prepare the track state
+      PM mask = PM::Predicted | PM::Jacobian | PM::Calibrated;
+      if (it != slBegin) {
+        // not the first TrackState, only need uncalibrated and calibrated
+        mask = PM::Calibrated;
+      }
+
+      ACTS_VERBOSE("Create temp track state with mask: " << mask);
+      // CAREFUL! This trackstate has a previous index that is not in this
+      // MultiTrajectory Visiting brackwards from this track state will
+      // fail!
+      auto ts = bufferTrajectory.makeTrackState(mask, prevTip);
```

Comment:
The only remaining reason for this `bufferTrajectory` is to allow the use of the previous measurement selector after calibration, which requires track states, right?

Reply:
- Removed buffer trajectory argument (in the only use case, the buffer trajectory and the main trajectory had been the same).
- Moved the buffer for newly created candidate track state proxies to the TrackStateCreator, where it is now a mutable member.

Reply:
moved temporary buffer for track state candidates back to the CKF results, which is now passed again by argument to the TrackStateCreator